solutions/azure-sqlmi-v1/nested/vm.json

```diff
@@ -0,0 +1,177 @@
+            ],
+            "metadata": {
+                "description": "OS disk type of the VM."
+            }
+        },
+        "securityType": {
+            "type": "string",
+            "defaultValue": "DiskWithVMGuestState",
+            "allowedValues": [
+                "VMGuestStateOnly",
+                "DiskWithVMGuestState"
+            ],
+            "metadata": {
+                "description": "VM security type."
+            }
+        }
+    },
+    "variables": {
+        "windowsConfiguration": {
+            "enableAutomaticUpdates": "true",
+            "provisionVmAgent": "true"
+        },
+        "vmName": "[concat(parameters('resourceName'), '-vm')]",
+        "networkInterfaceName": "[concat(parameters('resourceName'), '-vm-nic')]",
+        "networkSecurityGroupName": "[concat(parameters('resourceName'), '-vm-nsg')]",
+        "networkSecurityGroupId": "[resourceId(resourceGroup().name, 'Microsoft.Network/networkSecurityGroups', variables('networkSecurityGroupName'))]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Network/networkSecurityGroups",
+            "apiVersion": "2019-02-01",
+            "name": "[variables('networkSecurityGroupName')]",
+            "location": "[resourceGroup().location]",
+            "properties": {
+                "securityRules": [
+                    {
+                        "name": "RDP",
+                        "properties": {
+                            "priority": 100,
+                            "protocol": "Tcp",
+                            "access": "Allow",
+                            "direction": "Inbound",
+                            "sourceAddressPrefix": "*",
+                            "sourcePortRange": "*",
+                            "destinationAddressPrefix": "*",
+                            "destinationPortRange": "[3389]"
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "type": "Microsoft.Network/networkInterfaces",
+            "apiVersion": "2019-07-01",
+            "name": "[variables('networkInterfaceName')]",
+            "location": "[resourceGroup().location]",
+            "dependsOn": [
+                "[variables('networkSecurityGroupName')]"
+            ],
+            "properties": {
+                "ipConfigurations": [
+                    {
+                        "name": "ipConfigNode",
+                        "properties": {
+                            "privateIPAllocationMethod": "Dynamic",
+                            "subnet": {
+                                "id": "[parameters('subnetId')]"
+                            }
+                        }
+                    }
+                ],
+                "networkSecurityGroup": {
+                    "id": "[variables('networkSecurityGroupId')]"
+                }
+            }
+        },
+        {
+            "type": "Microsoft.Compute/virtualMachines",
```

Comment:
@unrealab could you please assist me in composing template for simple VM creation? I am getting following error with current config:
```
    "error": {
        "code": "BadRequest",
        "message": "Creation of VMs and Virtual Machine Scale Sets with ConfidentialVM security type is not supported in this region."
    }
```

Reply:
@milanbrkic-ms  - Confidential VMs are likely not available in all the regions, https://azure.microsoft.com/en-us/explore/global-infrastructure/products-by-region/?products=virtual-machines  this link has a table of what regions support Confidential VMs, You can look up by the Size of the VM you used. Ultimately, we'll want to restrict the possible regions the user can select based on the size of the VM you choose for the configuration. Let me know if you need more help.

Reply:
Got it!